src/ota/ota_handler.rs

```diff
@@ -142,18 +142,27 @@ impl OtaHandler {
         sdk: &impl Publisher,
         data: HashMap<String, AstarteType>,
     ) -> Result<(), DeviceManagerError> {
-        if let AstarteType::String(operation_str) = &data["operation"] {
-            match operation_str.parse::<OtaOperation>() {
-                Ok(OtaOperation::Update) => self.handle_update(sdk, data).await,
-                Ok(OtaOperation::Cancel) => self.handle_cancel(sdk, data).await,
-                Err(_) => Err(DeviceManagerError::OtaError(OtaError::Request(
+        let operation_str = match data.get("operation") {
+            Some(AstarteType::String(operation_str)) => operation_str,
+            _ => {
+                error!("missing ota operation: {:?}", data);
+
+                return Err(DeviceManagerError::OtaError(OtaError::Request(
```

Comment:
Here can use `let-else`
```suggestion
        let Some(AstarteType::String(operation_str)) = data.get("operation") else{
                error!("missing ota operation: {:?}", data);
                return Err(DeviceManagerError::OtaError(OtaError::Request(
```

Reply:
Done